docs/modules/password_auth_provider_callbacks.md

```diff
@@ -0,0 +1,155 @@
+# Password auth provider callbacks
+
+Password auth providers offer a way for server administrators to integrate
+their Synapse installation with an external authentication system. The callbacks can be
+registered by using the Module API's `register_password_auth_provider_callbacks` method.
+
+## Callbacks
+
+### `auth_checkers`
+
+```
+ auth_checkers: Dict[Tuple[str,Tuple], CHECK_AUTH_CALLBACK]
+```
+
+A dict mapping from tuples of a login type identifier (such as `m.login.password`) and a
+tuple of field names (such as `("password", "secret_thing")`) to authentication checking
+callbacks.
+
+The login type and field names are the things that should be provided by the user in their
+request to the `/login` API. 
+
+For example, if a module implements authentication checkers for two different login types: 
+-  `com.example.custom_login` 
+    - Expects `secret1` and `secret2` fields to be sent to `/login`
+    - Is checked by the method: `self.custom_login_check`
+- `m.login.password`
+    - Expects a `password` field to be sent to `/login`
+    - Is checked by the method: `self.password_checker` 
+    
+Then it should register the following dict:
+
+```python
+{ 
+    ("com.example.custom_login", ("secret1", "secret2")): self.check_custom_login,
+    ("m.login.password", ("password",)): self.password_checker,
+}
+```
```

Comment:
I feel like this could be all replaced with a link to https://matrix.org/docs/spec/client_server/latest#user-interactive-authentication-api (especially since we've got an example below on how it translates into code, so the additional example isn't strictly necessary)